Calibration/EcalCalibAlgos/src/ECALpedestalPCLHarvester.cc

```diff
@@ -0,0 +1,146 @@
+
+// system include files
+#include <memory>
+
+// user include files
+#include "Calibration/EcalCalibAlgos/interface/ECALpedestalPCLHarvester.h"
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+#include "DataFormats/EcalDetId/interface/EEDetId.h"
+#include "CondCore/DBOutputService/interface/PoolDBOutputService.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "CondFormats/EcalObjects/interface/EcalChannelStatus.h"
+#include "CondFormats/EcalObjects/interface/EcalChannelStatusCode.h"
+#include "CondFormats/DataRecord/interface/EcalPedestalsRcd.h"
+#include "CondFormats/DataRecord/interface/EcalChannelStatusRcd.h"
+#include "CommonTools/Utils/interface/StringToEnumValue.h"
+#include <iostream>
+#include <sstream>
+
+ECALpedestalPCLHarvester::ECALpedestalPCLHarvester(const edm::ParameterSet& ps):
+    currentPedestals_(0),channelStatus_(0){
+
+    chStatusToExclude_= StringToEnumValue<EcalChannelStatusCode::Code>(ps.getParameter<std::vector<std::string> >("ChannelStatusToExclude"));
+    minEntries_=ps.getParameter<int>("MinEntries");
+
+}
+
+void ECALpedestalPCLHarvester::dqmEndJob(DQMStore::IBooker& ibooker_, DQMStore::IGetter& igetter_) {
+    
+
+    // calculate pedestals and fill db record
+    EcalPedestals pedestals;
+
+    std::cout << "Filling ped record " << std::endl;
+       
+    for (uint16_t i =0; i< EBDetId::kSizeForDenseIndexing; ++i) {
+        std::stringstream hname;
```

Comment:
create the string stream outside of the loop and replace the next line by hname.str("EcalCalibration/EcalPedestalPCL/");